src/servers/ws-server.ts

```diff
@@ -42,151 +55,189 @@ const getContext = async (
+                level: ErrorLevel.Warn,
+              })
+              return [new GraphQLError("Only subscription operations are supported")]
+
+              // or if you want to be strict and terminate the connection on illegal operations
+              // throw new Error("Only subscription operations are supported")
+            }
+
+            // dont forget to validate
+            const errors = validate(args.schema, args.document)
+            if (errors.length > 0) {
+              // returning `GraphQLError[]` sends an `ErrorMessage` and stops the subscription
+              recordExceptionInCurrentSpan({
+                error: errors,
+                level: ErrorLevel.Warn,
+              })
+              return errors
+            }
+
+            // ready execution arguments
+            return args
+          },
+          onConnect: async (ctx) => {
+            baseLogger.debug("Connect", ctx)
+
+            const cookies = cookie.parse(ctx.extra.request.headers.cookie || "")
+            const hasCookies = typeof cookies !== "string" ? true : false
+            addAttributesToCurrentSpan({ "ws.hasCookies": hasCookies })
+            const kratosSessionCookie = hasCookies
+              ? cookies?.ory_kratos_session
+              : undefined
+
+            if (
+              typeof ctx.connectionParams?.Authorization !== "string" &&
+              !kratosSessionCookie
+            ) {
+              return true // anon connection ?
+            }
+
+            const context = await getContext(ctx)
+            if (typeof ctx.connectionParams?.Authorization === "string") {
+              authorizedContexts[ctx.connectionParams.Authorization] = context
+            }
+            if (kratosSessionCookie) {
+              authorizedContexts[kratosSessionCookie] = context
+            }
+
+            return true
+          },
+          context: async (ctx) => {
+            const cookies = cookie.parse(ctx.extra.request.headers.cookie || "")
+            const hasCookies = typeof cookies !== "string" ? true : false
+            const kratosSessionCookie = hasCookies
+              ? cookies?.ory_kratos_session
+              : undefined
+            // cookie auth
+            if (kratosSessionCookie) {
+              addAttributesToCurrentSpan({ "ws.authType": "cookie" })
+              return authorizedContexts[kratosSessionCookie]
+            }
+            // bearer auth
+            if (typeof ctx.connectionParams?.Authorization === "string") {
+              addAttributesToCurrentSpan({ "ws.authType": "bearer" })
+              return authorizedContexts[ctx.connectionParams?.Authorization]
+            }
+            // anon context
+            addAttributesToCurrentSpan({ "ws.authType": "anon" })
+            const context = await getContext(ctx)
+            return context
+          },
+          onNext: (ctx, msg, args, result) => {
+            baseLogger.debug("Next", { ctx, msg, args, result })
```

Comment:
this is probably where we should have some span events been register to honeycomb (in addition or replacement of the logger) if we want to have more meaning opentelemetry insights.